components/pages/Programs.tsx

```diff
@@ -13,27 +13,48 @@ import Filters from '@/components/general/Filters'
 import { CardProgram } from '@/components/cards'
 import { IconCheckCircle } from '@/components/icons'
 import programsContext from '@/context/programs/programsContext'
+import { useRouter } from 'next/router'
 
 
-const PagePrograms = ({ programs, mbaTypeOfProgram, mbaFormat }) => {
+const PagePrograms = ({programs, mbaTypeOfProgram, mbaFormat }) => {
+  const router = useRouter()
   const at = useAt()
 
-  const [currentField, setCurrentField] = useState(null)
+  let {currentFilter, setCurrentFilter} = useContext(programsContext)
+
 
   let fields
 
   if (at.profession || at.course) {
-    fields = programs.reduce((acc, curr) => {
-      if (!acc.includes(curr.study_field?.name))
-        acc.push(curr.study_field?.name)
-      return acc
-    }, [])
-
+    const studyFieldArr = [...new Set(programs.filter(item => item !== undefined && item?.study_field))]
```

Comment:
drop `item !== undefined`, leave out just `item`, would do falsy prove. Optional chaining doesn't work on the first level. Should be at least two levels deep.

```suggestion
    const studyFieldArr = [...new Set(programs.filter(item => item && item.study_field))]
```